pkg/system/phase2_creating.go

```diff
@@ -887,6 +887,14 @@ func (r *Reconciler) UpgradeMigrateDB() error {
 					c.Image = r.NooBaa.Status.ActualImage
 				}
 			}
+			for i := range podSpec.Containers {
+				c := &podSpec.Containers[i]
+				switch c.Name {
```

Comment:
Do we really need a switch here? Can't we do in simple if like in init-containers?
if c.Name == "db" {
	c.Image = options.DBMongoImage
}